File: backend/app/schemas/transaction.py
# app/schemas/transaction.py

from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from decimal import Decimal

class TransactionBase(BaseModel):
    category_id: int
    amount: Decimal
    description: Optional[str] = None

class TransactionCreate(TransactionBase):
    pass

class TransactionRead(TransactionBase):
    id: int
    user_id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/transaction.ts
// src/types/transaction.ts

export interface TransactionBase {
  category_id: number;
  amount: number;
  description?: string;
}

export interface TransactionCreate extends TransactionBase {}

export interface TransactionUpdate extends Partial<TransactionBase> {}

export interface CategoryRead {
  id: number;
  name: string;
  type: 'income' | 'expense';
}

export interface TransactionRead extends TransactionBase {
  id: number;
  user_id: number;
  created_at: string;      // ISO-date string
  category: CategoryRead;  // развёрнутая информация по категории
}
